src/IR/IR.hs

```diff
@@ -315,53 +320,100 @@ wellFormed (Prim   p    es   _) = wfPrim p es && all wellFormed es
-    arms = block bar (map arm as)
-    arm (a, e) = pretty a <+> drarrow <+> braces (pretty e)
-  pretty (Prim New   [r] t) = typeAnn t $ pretty "new" <+> pretty r
-  pretty (Prim Dup   [r] t) = typeAnn t $ pretty "__dup" <+> pretty r
-  pretty (Prim Drop  [r] t) = typeAnn t $ pretty "__drop" <+> pretty r
-  pretty (Prim Deref [r] t) = typeAnn t $ pretty "deref" <+> pretty r
-  pretty (Prim Assign [l, r] t) =
-    typeAnn t $ parens $ pretty l <+> pretty "<-" <+> braces (pretty r)
-  pretty (Prim After [d, l, r] t) = typeAnn t $ parens ae
+Adds 
+* indentation and line breaks
+* some parens (not minimal parens, but fewer than around every node)
+Omits 
+* let _ =
+* type annotations
+Reverts
+* curried funcs of one arg back to multiple arg funcs
+-}
+instance (Pretty t, TypeSystem t) => Pretty (Program t) where
+  pretty Program { programDefs = ds, typeDefs = tys } = tys' <> ds'
    where
-    ae =
-      pretty "after" <+> pretty d <> comma <+> pretty l <+> larrow <+> braces
-        (pretty r)
-  pretty (Prim Par es t) =
-    typeAnn t $ pretty "par" <+> block dbar (map pretty es)
-  pretty (Prim Wait es t) =
-    typeAnn t $ pretty "wait" <+> block dbar (map pretty es)
-  pretty (Prim Loop  [b] t) = typeAnn t $ pretty "loop" <+> braces (pretty b)
-  pretty (Prim Break []  t) = typeAnn t $ pretty "break"
-  pretty (Prim Return [e] t) =
-    typeAnn t $ pretty "return" <+> braces (pretty e)
-  pretty (Prim (PrimOp po) [l, r] t) =
-    parens $ pretty l <+> pretty po <+> pretty r <> pretty t
-  pretty (Prim p _ _) = error "Primitive expression not well-formed: " $ show p
+    ds' = case ds of
+      [h] -> prettyFuncDef h
+      _   -> vsep (map ((line <>) . prettyFuncDef) ds)
+    tys' = if null tys then pretty "" else vsep (prettyTypDef <$> tys)
+
+-- | Generates readable Doc representation of an IR Top Level Function
+prettyFuncDef :: (TypeSystem t, Pretty t) => (VarId, Expr t) -> Doc ann
+prettyFuncDef (v, l@(Lambda _ _ ty)) =
+  pretty v <+> typSig <+> pretty "=" <+> line <> indent 2 (fmt (void body))
+ where
+  typSig = hsep args <+> rarrow <+> pretty retTy
+  args   = zipWith (\arg t -> parens $ pretty arg <+> pretty ":" <+> pretty t)
+                   argIds
+                   argTys
+  (argIds, body ) = collectLambda l
+  (argTys, retTy) = collectArrow ty
+prettyFuncDef (v, e) = pretty v <+> pretty "=" <+> fmt (void e)
+
+-- | Generates readable Doc representation of an IR Type Definition
+prettyTypDef :: Pretty t => (TConId, TypeDef t) -> Doc ann
+prettyTypDef (tcon, TypeDef { variants = vars }) =
+  pretty "type"
+    <+> pretty tcon
+    <+> line
+    <>  indent indentNo (vsep $ map prettyDCon vars)
+    <>  line
+ where
+  prettyDCon :: (Pretty t) => (DConId, TypeVariant t) -> Doc ann
+  prettyDCon (dcon, VariantNamed args) =
+    pretty dcon <+> hsep (pretty . snd <$> args)
+  prettyDCon (dcon, VariantUnnamed args) =
+    pretty dcon <+> hsep (pretty <$> args)
+
+
+{- | Pretty prints IR Expr nodes without type annotations
```

Comment:
Yes, I tried to do that but I was getting a type error:
```
• Illegal instance declaration for ‘Pretty (Expr ())’
    (All instance types must be of the form (T a1 ... an)
     where a1 ... an are *distinct type variables*,
     and each type variable appears at most once in the instance head.
     Use FlexibleInstances if you want to disable this.)
```
I will enable flexible instances and make this pretty instead of fmtExpr.